odl/test/solvers/functional/default_functionals_test.py

```diff
@@ -516,5 +516,75 @@ def test_moreau_envelope_l2_sq(space, sigma):
                             x * 2 / (1 + 2 * sigma))
 
 
+def test_weighted_proximal(space):
+    """Test for the weighted proximal of a SeparableSum functional."""
+
+    l1 = odl.solvers.L1Norm(space)
+    l2 = odl.solvers.L2Norm(space)
+    func = odl.solvers.SeparableSum(l1, l2)
+
+    x = func.domain.one()
+
+    sigma = [0.5, 1.0]
+
+    prox = func.proximal(sigma)(x)
+    assert all_almost_equal(prox, [l1.proximal(sigma[0])(x[0]),
+                                   l2.proximal(sigma[1])(x[1])])
+
+
+def test_weighted_proximal_L2_norm_squared(space):
+    """Test for the weighted proximal of the squared L2 norm"""
+
+    # Define the functional on the space.
+    func = odl.solvers.L2NormSquared(space)
+
+    # Set the stepsize as a random element of the spaces
+    # with elements between 1 and 10.
+    sigma = odl.phantom.noise.uniform_noise(space, 1, 10)
```

Comment:
Obviously a bug here!

`uniform_noise` should be added here:

https://github.com/odlgroup/odl/blob/6b73551e2287dc80315d455eb14a53c88c2addbe/odl/phantom/noise.py#L17

and you should simply use `odl.phantom.uniform_noise`